users/tests/api/test_user.py

```diff
@@ -3,46 +3,79 @@
 from rest_framework.test import APIClient
 
 from users.models import User
+from users.tokens import verification_token
 
 
 @pytest.mark.django_db
-def test_register_user(test_social_profile):
+def test_register_user(test_social_profile, mocker):
     client = APIClient()
     data = {
         'email': 'test@profile.com',
         'password': 'password',
         'first_name': 'test',
         'last_name': 'profile',
     }
-    response = client.post('/api/auth/register/', data=data, format='json')
-    assert response.status_code == 200
-    assert response.data['key'] is not None
+    with mocker.patch.object(User, 'send_verification_email'):
+        response = client.post('/api/auth/register/', data=data, format='json')
+    assert response.status_code == 204
     user = User.objects.get(email='test@profile.com')
-    assert test_social_profile in user.social_profiles.all()
+    assert not user.is_verified
+    assert user.has_usable_password()
+    assert user.first_name == 'test' and user.last_name == 'profile'
+
+
+@pytest.mark.django_db
+def test_verify_email(test_user):
+    client = APIClient()
+    test_user.save()
+    data = {
+        'token': verification_token.make_token(test_user),
+        'email': test_user.email,
+    }
+    response = client.post('/api/auth/verify/', data=data, format='json')
+    test_user.refresh_from_db()
+    assert response.status_code == 204
+    assert test_user.is_verified
+
+
+@pytest.mark.django_db
+def test_verify_email__invalid_token(test_user):
+    client = APIClient()
+    other_user = User.objects.create_user(
+        username='other@test.com',
+        email='other@test.com',
+        password='password',
+    )
+    data = {
+        'token': verification_token.make_token(other_user),
+        'email': test_user.email
+    }
+    response = client.post('/api/auth/verify/', data=data, format='json')
+    assert response.status_code == 400
 
 
 @pytest.mark.django_db
-def test_checking_user_exists():
+def test_checking_user_exists(test_user):
```

Comment:
I'm of two minds w/ this type of test fixture. On the one hand it cleans up tests, but on the other hand it hides details that are useful for someone in the future that is unfamiliar with this test (in this case, the fact that the test_user has the email address `test@profile.com`). The fixture also couples tests together which can be frustrating when you need _kinda_ the fixture behavior but not quite the behavior. In this case it's pretty plain and simple, so maybe it doesn't matter. Decide what you think is best. I'm ok with either.

Reply:
I found myself needing the fixture enough that it felt better to separate it. I think this way is best for this particular case, but I do see your point and it is something to keep in mind for the future.

Reply:
👍